inginious/frontend/flask/mapping.py

```diff
@@ -154,3 +155,5 @@ def init_flask_mapping(flask_app):
                            view_func=AdministrationUserActionPage.as_view('administrationuseractionpage'))
     flask_app.add_url_rule('/<cookieless:sessionid>admin/<courseid>/api_tokens',
                            view_func=CourseAPITokensPage.as_view('courseapipage'))
+    flask_app.add_url_rule('/<cookieless:sessionid>admin/<courseid>/api_auth',
```

Comment:
So in our case we are considering tokens generated from the user interface are already authorized tokens. They should be used to access any API page. You therefore have to implement a common authorization check on top of every API page POST/GET processing methods.